hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RWQueueRpcExecutor.java

```diff
@@ -76,7 +76,9 @@ public RWQueueRpcExecutor(final String name, final int handlerCount, final int m
     numWriteQueues = calcNumWriters(this.numCallQueues, callqReadShare);
     writeHandlersCount = Math.max(numWriteQueues, calcNumWriters(handlerCount, callqReadShare));
 
-    int readQueues = calcNumReaders(this.numCallQueues, callqReadShare);
+    int readQueues = callqReadShare > 0 ?
```

Comment:
What is the return value of calcNumReaders if callqReadShare is 0?

Reply:
If `callqReadShare`==0, it will not be an instance of `RWQueueRpcExecutor`, then will not call `calcNumReaders`. But `readQueues` will be 0 under other circumstances by the origin design, when the `numCallQueues`=1 and `numWriteQueues`=1.

Reply:
What do you mean by "will not call calcNumReaders"? Either in the old code or in your new code, we will always call the calcNumReaders method?

Reply:
Thanks. I think when callqReadShare is 0, the RpcExecutor will be `FastPathBalancedQueueRpcExecutor` by default (see SimpleRpcScheduler line 88), not based on RWQueueRpcExecutor(which by default will create `FastPathRWQueueRpcExecutor`), and then there are not distinguished queues, all queues serve for read and write.